lib/resty/healthcheck.lua

```diff
@@ -919,8 +960,32 @@ function checker:run_single_check(ip, port, hostname, hostheader)
   end
 
   local req_headers = self.checks.active.headers
-  local headers = table.concat(req_headers, "\r\n")
-  if #headers > 0 then
+  local headers
```

Comment:
Are `self.checks.active.headers` ever mutated after the healthcheck is created? If not, maybe we should consider caching the final headers string somewhere in `self` now that we're doing a lot more work to create it.